analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RoslynSymbolicExecution.cs

```diff
@@ -74,7 +74,12 @@ public void Execute()
                 return;
             }
             var current = queue.Dequeue();
-            if (visited.Add(current) && current.AddVisit() <= MaxOperationVisits)
+            if (visited.Add(current)
+                && current.AddVisit() is var visitCount
+                && (visitCount <= MaxOperationVisits
+                || (visitCount == MaxOperationVisits + 1
+                    && current.Block.ConditionalSuccessor is not null
+                    && syntaxClassifier.IsInLoopCondition(current.Block.BranchValue?.Syntax))))
```

Comment:
Why is this needed instead of increasing the constant that is designed to control the amount?

Reply:
We need `MaxOperationVisits` for regular operations and `MaxOperationVisits + 1` for loop conditions.